File: testdata/fixtures/bench/go/vulnerable/a03_command_injection.go
// Source: CWE-78 - OS Command Injection in Go
// Expected: BATOU-INJ-002 (Command Injection via exec.Command)
// OWASP: A03:2021 - Injection (OS Command Injection)

package handler

import (
	"fmt"
	"net/http"
	"os/exec"
)

func PingHost(w http.ResponseWriter, r *http.Request) {
	host := r.URL.Query().Get("host")
	cmd := exec.Command("sh", "-c", "ping -c 4 "+host)
	output, err := cmd.CombinedOutput()
	if err != nil {
		http.Error(w, fmt.Sprintf("Error: %s", err), http.StatusInternalServerError)
		return
	}
	w.Write(output)
}

func ConvertFile(w http.ResponseWriter, r *http.Request) {
	inputPath := r.FormValue("input")
	outputFormat := r.FormValue("format")
	cmdStr := fmt.Sprintf("ffmpeg -i %s output.%s", inputPath, outputFormat)
	cmd := exec.Command("bash", "-c", cmdStr)
	output, err := cmd.CombinedOutput()
	if err != nil {
		http.Error(w, string(output), http.StatusInternalServerError)
		return
	}
	w.Write([]byte("Conversion complete"))
}

File: testdata/fixtures/bench/c/vulnerable/a03_command_injection.c
// Source: CWE-78 - OS Command Injection in C
// Expected: BATOU-MEM-001 (Banned Function - sprintf), BATOU-MEM-011 (system with variable), BATOU-MEM-010 (popen with variable)
// OWASP: A03:2021 - Injection (OS Command Injection)

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void ping_host(const char *hostname) {
    char command[256];
    sprintf(command, "ping -c 4 %s", hostname);
    system(command);
}

void convert_file(const char *input_path, const char *format) {
    char cmd[512];
    snprintf(cmd, sizeof(cmd), "ffmpeg -i %s output.%s", input_path, format);
    system(cmd);
}

int check_dns(const char *domain) {
    char buf[256];
    sprintf(buf, "nslookup %s", domain);
    FILE *fp = popen(buf, "r");
    if (!fp) return -1;
    char result[1024];
    while (fgets(result, sizeof(result), fp)) {
        printf("%s", result);
    }
    return pclose(fp);
}
